src/Sound/Tidal/UI.hs

```diff
@@ -21,6 +19,22 @@ module Sound.Tidal.UI where
     along with this library.  If not, see <http://www.gnu.org/licenses/>.
 -}
 
+{-|
+    This module provides the main user interface functions, including sources
+    of randomness and transformations of patterns. All these functions are available
+    in the context of the TidalCycles REPL.
+
+    Many functions in this module taking 'Pattern' values as arguments have a
+    corresponding function with an underscore prepended to its name (e.g.
+    'degradeBy' and '_degradeBy'). These functions take plain old data types,
+    not 'Pattern's, and are mainly useful if you're getting errors about
+    ambiguous types. In some cases, they may be more efficient than their
+    equivalents that take 'Pattern' values.
```

Comment:
I'd always assumed that the underscore variants were essentially internal implementations and not intended for public use. If that's the case, then we shouldn't be encouraging their use, and should eventually exclude them from export lists. @yaxu, is that right, or am I wrong and these variants should be considered part of the public API?

Are there cases where you've run into ambiguous type errors?

Reply:
If memory serves, I once had enabled perhaps too many `{-# LANGUAGE #-}` pragmas in the Tidal REPL and got out of it by using an underscore variant. I’m happy to elide this section and just replace it with “the underscore variants are private,” if @yaxu thinks that’s appropriate.

Reply:
Hm well I think they shouldn't be totally private, as they're useful in writing custom functions. I would say something like, the underscore variants generally only accept plain values and not patterns apart from their final argument, and are intended mainly for those developing or extending tidal.

Reply:
@yaxu Gotcha. I’ve changed the wording to reflect this.